test/validation/api/packet/packet.c

```diff
@@ -929,13 +929,16 @@ static void _verify_headroom_shift(odp_packet_t *pkt,
 	CU_ASSERT_PTR_NOT_NULL(data);
 	if (extended) {
 		CU_ASSERT(rc >= 0);
-		if (shift >= 0) {
-			CU_ASSERT(odp_packet_seg_len(*pkt) == shift - room);
-		} else {
-			CU_ASSERT(odp_packet_headroom(*pkt) >=
-				  (uint32_t)abs(shift) - seg_data_len);
+		if (rc == 0) {
+			if (shift >= 0) {
+				CU_ASSERT(odp_packet_seg_len(*pkt) ==
+					  shift - room);
+			} else {
+				CU_ASSERT(odp_packet_headroom(*pkt) >=
+					  (uint32_t)abs(shift) - seg_data_len);
+			}
+			CU_ASSERT(odp_packet_head(*pkt) != head_orig);
```

Comment:
So what exactly is the problem that you are fixing by not executing these lines if the packet changed in the extend or trunc call? These operations are not using the old handle but the new handle returned by the extend/trunk call.

Reply:
Spec says. 
```
"A successful operation overwrites the packet handle with a new handle, which
application must use as the reference to the packet instead of the old
handle. Depending on the implementation, the old and new handles may be
equal.
The operation return value indicates if any packet data or metadata (e.g.
user_area) were moved in memory during the operation. If some memory areas
were moved, application must use new packet/segment handles to update
data pointers. Otherwise, all old pointers remain valid.".

``` 
I believe by this it means the packet pointed to by the new handle can have an entirely different layout from the original and it is left to the implementation. So we can't compare any property of the original packet with the new packet.

Assuming my understanding of the spec is correct, 

1. seg_data_len is a property of the original packet. 
If suppose the new implementation simply adds a new segment at the beginning of the packet with the required extend length, it means the new packet can have different segmentation and new headroom need not be equal to (uint32_t)abs(shift) - seg_data_len.

2. head_orig is a property of the original packet. 
If suppose the implementation simply memmove the existing data ahead in the  segment to create more space at the beginning which means we cannot guarantee CU_ASSERT(odp_packet_head(*pkt) != head_orig);

Reply:
> I believe by this it means the packet pointed to by the new handle can have an entirely different layout from the original and it is left to the implementation. So we can't compare any property of the original packet with the new packet.

That is what I think too *). The problem with the code is not that it does not realize that the handle might change but the issue that the packet layout may change. The commit message could be made more clear what the real problem is.

*) There are, of course, some properties that can be compared and that maybe should be checked in the validation code. The old packet data should still be there (even if maybe in a new memory location), packet length should change exactly the requested amount, etc.

Reply:
Right. Anything relating to the original packet layout in the memory should not be compared.

Packet length needs to be verified to be orig_pkt_len + extend_len which it is doing and I have not removed that check from the existing code.

Data within the packet also must remain intact. But its not verifying the same in the existing code. So I didn't add that additional verification step.

I hope the change is fine. I will modify the commit message to be more descriptive explaining the issue at hand.